tests/functional/iceberg/test_table_basic.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+
+from pathlib import Path
+
+from dbt.tests.util import run_dbt, rm_file
+
+_MODEL_BASIC_TABLE_MODEL = """
+{{
+  config(
+    materialized = "table",
+    cluster_by=['id'],
+  )
+}}
+select 1 as id
+""".strip()
+
+_MODEL_BASIC_ICEBERG_MODEL = """
+{{
+  config(
+    transient = "true",
+    materialized = "table",
+    cluster_by=['id'],
+    table_format="iceberg",
+    external_volume="s3_iceberg_snow",
+    base_location_subpath="subpath",
+  )
+}}
+
+select * from {{ ref('first_table') }}
+""".strip()
+
+_MODEL_BUILT_ON_ICEBERG_TABLE = """
+{{
+  config(
+    materialized = "table",
+  )
+}}
+select * from {{ ref('iceberg_table') }}
+""".strip()
+
+_MODEL_TABLE_FOR_SWAP = """
+{{
+  config(
+    materialized = "table",
+  )
+}}
+select 1 as id
+""".strip()
+
+_MODEL_TABLE_FOR_SWAP_ICEBERG = """
+{{
+  config(
+    materialized = "table",
+    table_format="iceberg",
+    external_volume="s3_iceberg_snow",
+    base_location_subpath="subpath",
+  )
+}}
+select 1 as id
+""".strip()
+
+
+class TestIcebergTableBuilds:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "first_table.sql": _MODEL_BASIC_TABLE_MODEL,
+            "iceberg_table.sql": _MODEL_BASIC_ICEBERG_MODEL,
+            "table_built_on_iceberg_table.sql": _MODEL_BUILT_ON_ICEBERG_TABLE,
+        }
+
+    def test_iceberg_tables_build_and_can_be_referred(self, project):
+        run_results = run_dbt()
+        assert len(run_results) == 3
+
+
+class TestIcebergTableTypeBuildsOnExistingTable:
+    model_name = "my_model.sql"
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {self.model_name: _MODEL_TABLE_FOR_SWAP}
+
+    def test_changing_model_types(self, project):
```

Comment:
Can we add a view to this swapping test too? I think it would work since we probably just check if it's not a view and drop it, but it was scenarios like that which caught me on DTs and MVs.

Reply:
You got it. Added!